### PR TITLE
Change sender and receiver type

### PR DESCRIPTION
**Pull request by popenta**
Changed the `sender`,`receiver` and `guardian` types as described in the [specs](https://github.com/multiversx/mx-sdk-specs/blob/main/core/transaction.md).

### CODE REVIEW

File: multiversx_sdk/controllers/interfaces.py [hunk 1]
```diff
@@ -1,12 +1,19 @@
 from typing import Any, List, Optional, Protocol, Union
 
-from multiversx_sdk.core.interfaces import IAddress
 from multiversx_sdk.core.smart_contract_query import (
     SmartContractQuery, SmartContractQueryResponse)
 from multiversx_sdk.core.transaction_on_network import TransactionOnNetwork
 from multiversx_sdk.network_providers.resources import AwaitingOptions
 
 
+class IAddress(Protocol):
```
Comment (andreibancioiu) [inline multiversx_sdk/controllers/interfaces.py:9]: Maybe discard this, as well?

File: multiversx_sdk/controllers/smart_contract_controller.py [hunk 2]
```diff
@@ -52,21 +53,23 @@ def create_transaction_for_deploy(self,
         )
 
         transaction.nonce = nonce
-        transaction.signature = sender.sign(self.tx_computer.compute_bytes_for_signing(transaction))
+        transaction.signature = sender.sign(
```
Comment (andreibancioiu) [inline multiversx_sdk/controllers/smart_contract_controller.py:56]: Line is broken / formatting issue?

File: multiversx_sdk/wallet/user_test.py [hunk 3]
```diff
@@ -119,8 +119,8 @@ def test_sign_transaction():
     tx = Transaction(
         nonce=89,
         value=0,
-        receiver="erd1spyavw0956vq68xj8y4tenjpq2wd5a9p2c6j8gsz7ztyrnpxrruqzu66jx",
-        sender="erd1qyu5wthldzr8wx5c9ucg8kjagg0jfs53s8nr3zpz3hypefsdd8ssycr6th",
+        receiver=Address.new_from_bech32("erd1spyavw0956vq68xj8y4tenjpq2wd5a9p2c6j8gsz7ztyrnpxrruqzu66jx"),
```
Comment (andreibancioiu) [inline multiversx_sdk/wallet/user_test.py:122]: Make sure to update the specs on this matter, as well.

How will we handle this in JS?

File: multiversx_sdk/core/transaction_test.py [hunk 4]
```diff
@@ -361,6 +359,9 @@ def test_transaction_converter(self):
         )
 
         tx_as_dict = transaction.to_dictionary()
+        print(transaction.__dict__)
         restored_tx = Transaction.new_from_dictionary(tx_as_dict)
+        print("------------")
```
Comment (andreibancioiu) [inline multiversx_sdk/core/transaction_test.py:364]: Debugging artifact.

File: multiversx_sdk/core/transaction_computer.py [hunk 5]
```diff
@@ -11,16 +11,20 @@
...
 
 
 class TransactionComputer:
     def __init__(self) -> None:
         pass
 
-    def compute_transaction_fee(self, transaction: ITransaction, network_config: INetworkConfig) -> int:
-        move_balance_gas = network_config.min_gas_limit + len(transaction.data) * network_config.gas_per_data_byte
+    def compute_transaction_fee(self, transaction: Transaction, network_config: INetworkConfig) -> int:
+        move_balance_gas = network_config.min_gas_limit + \
```
Comment (andreibancioiu) [inline multiversx_sdk/core/transaction_computer.py:26]: Possible formatting issue / line breaking.

File: multiversx_sdk/core/account.py [hunk 6]
```diff
@@ -1,22 +1,46 @@
...
+        signer = UserSigner.from_pem_file(file_path, index)
+        return Account(signer, hrp)
+
+    @classmethod
+    def new_from_keystore(cls,
+                          file_path: Path,
+                          password: str,
+                          address_index: int = 0,
+                          hrp: str = DEFAULT_HRP) -> "Account":
+        secret_key = UserWallet.load_secret_key(
```
Comment (andreibancioiu) [inline multiversx_sdk/core/account.py:26]: Broken line,

File: multiversx_sdk/controllers/interfaces.py (hunk 1 above)
Comment (popenta) [reply multiversx_sdk/controllers/interfaces.py:9]: Yep, deleted it.

File: multiversx_sdk/controllers/smart_contract_controller.py (hunk 2 above)
Comment (popenta) [reply multiversx_sdk/controllers/smart_contract_controller.py:56]: fixed

**Review by danielailie — DISMISSED** [review]

File: multiversx_sdk/wallet/user_test.py (hunk 3 above)
Comment (popenta) [reply multiversx_sdk/wallet/user_test.py:122]: If you refer to the `Transaction` specs, they are updated [here](https://github.com/multiversx/mx-sdk-specs/blob/main/core/transaction.md). Regarding handling this in the js sdk, I am not sure what you mean.

File: multiversx_sdk/core/transaction_test.py (hunk 4 above)
Comment (popenta) [reply multiversx_sdk/core/transaction_test.py:364]: removed

File: multiversx_sdk/core/transaction_computer.py (hunk 5 above)
Comment (popenta) [reply multiversx_sdk/core/transaction_computer.py:26]: fixed

File: multiversx_sdk/core/account.py (hunk 6 above)
Comment (popenta) [reply multiversx_sdk/core/account.py:26]: fixed

**Review by danielailie — APPROVED** [review]

**Review by danielailie — APPROVED** [review]

**Review by andreibancioiu — APPROVED** [review]

**Pull request merged**